### PR TITLE
Add mapping url that returns block number for the given slot number.

### PR DESCRIPTION
**Pull request by 0xSasaPrsic**
Add url that maps given slot to block number. It uses beaconcha api to fetch block info.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -32,6 +32,7 @@ struct AppState {
     ethereum_client: HttpClient,
     succinct_client: Client,
     succinct_base_url: String,
+    beaconcha_url: String,
```
Comment (QEDK) [inline src/main.rs:35]: Should be `beaconchain_base_url` I think `.in` doman is part of the service's name.

File: src/main.rs [hunk 2]
```diff
@@ -257,6 +273,62 @@ async fn get_avl_proof(
...
 }
 
+/// Creates a request to the beaconcha service for mapping slot to the block number.
+#[inline(always)]
+async fn get_beacon_slot(
+    Path(slot): Path<U256>,
+    State(state): State<Arc<AppState>>,
+) -> impl IntoResponse {
+    let resp = state
+        .succinct_client
```
Comment (QEDK) [inline src/main.rs:283]: Let's rename `succinct_client` to `reqwest_client`?

File: src/main.rs [hunk 3]
```diff
@@ -66,6 +67,21 @@ struct SuccinctAPIResponse {
     success: Option<bool>,
 }
 
+#[derive(Deserialize)]
+struct BeaconAPIResponse {
+    status: String,
+    data: BeaconAPIResponseData,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
```
Comment (QEDK) [inline src/main.rs:76]: Any reason we need `Debug`?

**Review by QEDK — CHANGES_REQUESTED** [review]
Some minor changes, lmk.

File: src/main.rs (hunk 3 above)
Comment (0xSasaPrsic) [reply src/main.rs:76]: Not needed, deleting.

**Review by QEDK — APPROVED** [review]
LGTM.

**Pull request merged**